### PR TITLE
test: more tests and improve coverage

### LINKED ISSUE #23: More tests!
**Is your feature request related to a problem? Please describe.**

No integration tests mean wide ranging can introduce bugs and regressions not caught.

**Describe the solution you'd like**

💯 coverage!

**Describe alternatives you've considered**

No bragging rights for 💯 coverage

**Additional context**

N/A

### PR DESCRIPTION
**Pull request by iamogbz**
### Description

Resolves #23 

- [x] Missing snapshots
- [x] Written snapshots
- [x] Unused snapshots
- [x] Removed snapshots
- [x] Deleted snapshot files

### CODE REVIEW

File: tests/test_integration.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    return testdir.runpytest("-v", "--snapshot-update"), testdir, tests, filepath
+
+
+def test_fixture(stubs):
+    result = stubs[0]
+    result.stdout.fnmatch_lines(["*::test_injection PASSED*"])
+    assert result.ret == 0
+
+
+def test_generate_snapshots(stubs):
```
Comment (iamogbz) [inline tests/test_integration.py]:
```suggestion
def test_generated_snapshots(stubs):
```

File: tests/test_integration.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        "unused": """def test_unused(snapshot):
+    assert snapshot == 'unused'""",
+    }
+    pyfile_content = "\n\n".join(tests.values())
+    testdir.makepyfile(test_file=pyfile_content)
+    filepath = os.path.join(testdir.tmpdir, "__snapshots__", "test_file.yaml")
+    return testdir.runpytest("-v", "--snapshot-update"), testdir, tests, filepath
+
+
+def test_fixture(stubs):
```
Comment (iamogbz) [inline tests/test_integration.py]:
```suggestion
def test_injected_fixture(stubs):
```

File: tests/test_integration.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import os
+
+import pytest
+
+
+@pytest.fixture
+def stubs(testdir):
+    tests = {
+        "inject": """def test_injection(snapshot):
```
Comment (noahnu) [inline tests/test_integration.py]: might be easier to read if we start it on a newline:
```python
"inject": (
    """
    def test_injection(snapshot):
        assert snapshot is not None
    """
)
```

and then trim space if necessary

Comment (iamogbz) [reply tests/test_integration.py]: Done

File: tests/test_snapshots.py [hunk 4]
```diff
@@ -21,6 +21,11 @@ def test_multiple_snapshots(snapshot):
     assert "Third." == snapshot
 
 
+def test_missing_snapshots(snapshot):
+    with pytest.raises(AssertionError, match="Snapshot does not exist"):
+        assert "This snapshot should not be written" == snapshot
```
[the remark this answers is not in the document]
Comment (iamogbz) [reply tests/test_snapshots.py]: If you run it with the update flag, it produces a snapshot but then the test would fail

**Comment by iamogbz** [comment]
@all-contributors please add @iamogbz and @noahnu for tests

**Pull request merged**